src/transformers/models/mm_grounding_dino/modular_mm_grounding_dino.py

```diff
@@ -203,16 +319,83 @@ def _init_weights(self, module):
             nn.init.constant_(module.bias, -math.log((1 - 0.01) / 0.01))
 
 
-# TODO: this one is useless, but without it class order in modeling gets messed up
+class MMGroundingDinoConvEncoder(GroundingDinoConvEncoder):
+    pass
+
+
+class MMGroundingDinoConvModel(GroundingDinoConvModel):
+    pass
+
+
+class MMGroundingDinoEncoder(GroundingDinoEncoder):
+    pass
+
+
 class MMGroundingDinoDecoder(GroundingDinoDecoder):
     pass
 
 
-class MMGroundingDinoModel(GroundingDinoModel):
+class MMGroundingDinoModel(GroundingDinoModel, MMGroundingDinoPreTrainedModel):
     def __init__(self, config: MMGroundingDinoConfig):
-        super().__init__(config)
+        MMGroundingDinoPreTrainedModel.__init__(config)
+
+        # Create backbone + positional encoding
+        backbone = MMGroundingDinoConvEncoder(config)
+        position_embeddings = build_position_encoding(config)
+        self.backbone = MMGroundingDinoConvModel(backbone, position_embeddings)
+
+        # Create input projection layers
+        if config.num_feature_levels > 1:
```

Comment:
Same here, can we remove this if/else? I suppose `num_feature_levels` > 1 for all checkpoints